hypothesis.py

```diff
@@ -96,59 +150,39 @@ def fraud_metrics(sample_data: pd.DataFrame,
                               mean and variance in isoscapes.
     p_value_target: desired p_value for the t-test (e.sample_data: 0.05)
     '''
-    sample_data = sample_data.groupby([
-        _TREE_CODE_COLUMN_NAME,
-        _LONGITUDE_COLUMN_NAME,
-        _LATITUDE_COLUMN_NAME,
-        _FRAUDULENT_COLUMN_NAME])[isotope_column_names]
-
-    # Counts the number of locations in the sample have more than one row in dataset.
-    rows = 0
-
-    true_positive = 0
-    true_negative = 0
-    false_positive = 0
-    false_negative = 0
-    for group_key, isotope_values in sample_data:
-      if isotope_values.shape[0] <= 1:
-        continue
-
-      p_values = []
-      for i, isotope_column_name in enumerate(isotope_column_names):
-        hypothesis_test = sample_ttest(group_key[1],
-                                       group_key[2],
-                                       isotope_values[isotope_column_name],
-                                       means_isoscapes[i],
-                                       variances_isoscapes[i],
-                                       sample_size_per_location,
-                                       p_value_target)
-        p_values.append(hypothesis_test.p_value)
-      combined_p_value = np.array(p_values).prod()
-
-      if not group_key[3]:
-        if combined_p_value >= p_value_target:
-          true_negative += 1
-        else:
-          false_positive += 1
-      else:
-        if combined_p_value >= p_value_target:
-          false_negative += 1
-        else:
-          true_positive += 1
-
-      rows += 1
-
+    predictions = get_predictions(sample_data,
+                  isotope_column_names,
+                  means_isoscapes,
+                  variances_isoscapes,
+                  sample_size_per_location)
+    y_true = predictions['fraud']
+    y_pred = predictions['predicted_fraud']
+    
+    true_positives = (
+      predictions[(predictions['fraud'] == True) &
```

Comment:
I notice there's only a single ampersand here. That is a bitwise operator, not a logical AND. Would it be possible to use the logical operator instead?

Reply:
Ack, changed.